src/client/java/teammates/client/scripts/scalabilitytests/InstructorFeedbackResultsPageScalabilityTest.java

```diff
@@ -18,20 +25,150 @@
 
     private static final String DATA_FOLDER_PATH = "src/client/java/teammates/client/scripts/scalabilitytests/data/";
     private static final Logger log = Logger.getLogger();
+    private static DataBundle testDataMax;
 
     @Override
     protected void prepareTestData() {
         // the actual test data is refreshed before each test method
     }
 
     private void refreshTestData(int numStudents, int numQuestions) {
-        refreshTestData(
-                "InstructorFeedbackResultsPageScaleTest-" + numStudents + "Students" + numQuestions + "Questions.json");
+
+        // modify set of students for each test according to its requirements
+        if (testData.students.size() == 0) { // if current collection of students is empty
+            addStudents(numStudents);
+        } else if (testData.students.size() < numStudents) { // current number of students is below required
+            addStudents(numStudents - testData.students.size());
+        } else if (testData.students.size() > numStudents) { // current number of students is above required
+            decreaseNumOfStudents(numStudents);
+        }
+
+        // modify set of questions for each test according to its requirements
+        if (testData.feedbackQuestions.size() == 0) { // if current collection of questions is empty
+            addQuestions(numQuestions);
+        } else if (testData.feedbackQuestions.size() < numQuestions) { // current number of questions is below required
+            addQuestions(numQuestions - testData.feedbackQuestions.size());
+        } else if (testData.feedbackQuestions.size() > numQuestions) { // current number of questions is above required
+            decreaseNumOfQuestions(numQuestions);
+        }
+
+        // modify set of responses for each test according to its sets of students and questions
+        updateFeedbackResponses();
+
+        // perform test by reading each entity from datastore or persisting if absent
+        verifyOrPersistTestDataToDatastore();
+    }
+
+    // verify if entities for testing already exist in datastore
+    private void verifyOrPersistTestDataToDatastore() {
+        for (StudentAttributes student : testData.students.values()) {
+            try {
+                verifyPresentInDatastore(student);
```

Comment:
Yes, I understood my mistake now - I should check every test data if it exists in datastore. So I corrected 3rd issue - added checks for not only for ```Instructor```, but also for ```Account``` and ```Course```.  As for feedbackSession - there is a method in ```BackDoor``` to check it in datastor, but there isn't a method ```createFeedbackSession()``` - I would try to create but not really sure how to write ```createFeedbackSession()``` correctly (it seems to be more complicated that creating ```Student``` or ```Question```).

Also, I cannot access to ```getEntity()``` method from here, because it is private in class ```BaseTestCaseWithDatastoreAccess``` - will it be OK to make it public? 
Or maybe solving that 3rd issue also solves 1st and 2nd issues from your comments?
Could you please check now? And help to write ```createFeedbackSession()``` in ```BackDoor```? 
Thanks!